js/views/modals/purchase/Purchase.js

```diff
@@ -140,19 +166,51 @@ export default class extends BaseModal {
   }
 
   purchaseListing() {
+    // if the listing has shipping, set it
+    if (this.listing.get('shippingOptions') && this.listing.get('shippingOptions').length) {
```

Comment:
It's possible for the user to delete all of their addresses in settings before trying to make a purchase, so there's a check here to make sure that doesn't happen. The order model can't do the checking because it doesn't know if the order needs an address or not, so it always treats the address data as optional.